container/test-impl-base/src/main/java/org/jboss/arquillian/container/test/impl/client/ContainerEventController.java

```diff
@@ -136,12 +138,11 @@ private void createContext(EventContext<? extends TestEvent> context) {
     * TODO: This should not rely on direct Reflection, but rather access the metadata through some
     * common metadata layer.
     */
-
     private void lookup(Method method, ResultCallback callback) {
         DeploymentTargetDescription deploymentTarget = locateDeployment(method);
 
         ContainerRegistry containerRegistry = this.containerRegistry.get();
-        DeploymentScenario deploymentScenario = this.deploymentScenario.get();
+        DeploymentScenario deploymentScenario = Objects.requireNonNull(this.deploymentScenario.get(), "deploymentScenario cannot be null");
```

Comment:
I'm not sure what we gain here.

Reply:
not much I was using this as a marker as the main issue I have with parallel tests is the NPE here

Reply:
Ah okay. I would think then that there is likely an issue activating the context in a multi-threaded environment. That's typically when `null` would be returned there.